File: problems/atcoder/abc088_b/main.cpp
/**
 * Author by Rainboy blog: https://rainboylv.com github: https://github.com/rainboylvx
 * rbook: -> https://rbook.roj.ac.cn  https://rbook2.roj.ac.cn
 * rainboy的学习导航网站: https://idx.roj.ac.cn
 * create_at: 2026-07-10 16:11
 * update_at: 2026-07-10 16:11
 */
#include <bits/stdc++.h>
using namespace std;

int main() {
    ios::sync_with_stdio(false);
    cin.tie(nullptr);

    int n; cin >> n;
    vector<int> a(n);
    for (int i = 0; i < n; i++) cin >> a[i];

    sort(a.rbegin(), a.rend());

    int alice = 0, bob = 0;
    for (int i = 0; i < n; i++) {
        if (i % 2 == 0) alice += a[i];
        else            bob   += a[i];
    }
    cout << alice - bob << '\n';

    return 0;
}

File: problems/atcoder/abc088_b/main.hs
{-
 Author by Rainboy blog: https://rainboylv.com github: https://github.com/rainboylvx
 rbook: -> https://rbook.roj.ac.cn  https://rbook2.roj.ac.cn
 rainboy的学习导航网站: https://idx.roj.ac.cn
 create_at: 2026-07-10 16:16
 update_at: 2026-07-10 16:16
-}
import Data.List (sort)

main = do
    input <- getContents
    let (_:xs) = map read (words input) :: [Int]
        ys = reverse (sort xs)
        alice = sum [ y | (i,y) <- zip [0..] ys, even i ]
        bob   = sum [ y | (i,y) <- zip [0..] ys, odd  i ]
    print $ alice - bob
